src/main/java/baekjoon/barkingdog/bfs/Gold2146.kt

```
@@ -0,0 +1,89 @@
+                visit[row][column] = true
+                map[row][column] = islandId
+
+                while (queue.isNotEmpty()) {
+                    val current = queue.removeFirst()
+
+                    for ((rowDirection, columnDirection) in directions) {
+                        val nextRow = current.first + rowDirection
+                        val nextColumn = current.second + columnDirection
+
+                        if (nextRow in 0..<size && nextColumn in 0..<size &&
+                            map[nextRow][nextColumn] == 1 && !visit[nextRow][nextColumn]) {
```

detekt (1.23.8)

[warning] This condition is too complex (4). Defined complexity threshold for conditions is set to '4'

(detekt.complexity.ComplexCondition)